src/foundry/common/primitives/set.mjs.d.ts

```diff
@@ -0,0 +1,99 @@
+interface Set<T> {
+  /**
+   * Return the difference of two sets.
+   * @param other - Some other set to compare against
+   * @returns The difference defined as objects in this which are not present in other
+   */
+  difference(other: Set<T>): Set<T>;
+
+  /**
+   * Test whether this set is equal to some other set.
+   * Sets are equal if they share the same members, independent of order
+   * @param other - Some other set to compare against
+   * @returns Are the sets equal?
+   */
+  equals(other: Set<T>): boolean;
+
+  /**
+   * Return the first value from the set.
+   * @returns The first element in the set, or undefined
+   */
+  first(): T | undefined;
+
+  /**
+   * Return the intersection of two sets.
+   * @param other - Some other set to compare against
+   * @returns The intersection of both sets
+   */
+  intersection(other: Set<T>): Set<T>;
+
+  /**
+   * Test whether this set has an intersection with another set.
+   * @param other - Another set to compare against
+   * @returns Do the sets intersect?
+   */
+  intersects(other: Set<T>): boolean;
+
+  /**
+   * Test whether this set is a subset of some other set.
+   * A set is a subset if all its members are also present in the other set.
+   * @param other - Some other set that may be a subset of this one
+   * @returns Is the other set a subset of this one?
+   */
+  isSubset(other: Set<T>): boolean;
+
+  /**
+   * Convert a set to a JSON object by mapping its contents to an array
+   * @returns The set elements as an array.
+   */
+  toObject(): T[];
+
+  /**
+   * Test whether every element in this Set satisfies a certain test criterion.
+   * @see {@link Array#every}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being tested.
+   * @returns Does every element in the set satisfy the test criterion?
+   */
+  every(test: (value: T, index: number, set: Set<T>) => boolean): boolean;
+
+  /**
+   * Filter this set to create a subset of elements which satisfy a certain test criterion.
+   * @see {@link Array#filter}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being filtered.
+   * @returns A new Set containing only elements which satisfy the test criterion.
+   */
+  filter(test: (value: T, index: number, set: Set<T>) => boolean): Set<T>;
+
+  /**
+   * Find the first element in this set which satisfies a certain test criterion.
+   * @see {@link Array#find}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being searched.
+   * @returns The first element in the set which satisfies the test criterion, or undefined.
+   */
+  find(test: (value: T, index: number, set: Set<T>) => boolean): T | undefined;
```

Comment:
```suggestion
  find<S extends T>(test: (value: T, index: set: Set<T>) => value is S): S | undefined;
  find(test: (value: T, index: number, set: Set<T>) => unknown): T | undefined;
```

Reply:
You removed the index position type somehow.

Reply:
Oops